### DISCUSSION TITLE
How to deploy correctly when changing environment variables

### DISCUSSION THREAD

**Discussion by AymaneHan1** [microsoft/sample-app-aoai-chatGPT#435] [Q&A]
Hello!

I want to add some changes to the environment variables and deploy it. The problem is env vars are not taken in account when I deploy the the app. I tested changing the title of the front and it is deployed correctly. But when I change the ens vars like max tooken to 2, I can't have see the modification once deployed. 

Can someone help me this please?  Thank you in advance

**Comment by ntabernacle** [comment]
One solution could be to SSH into your virtual machine and update the variables there.

**Comment by rhematt** [comment]
the workflow I follow is thus:

1. Deploy from portal
2. update the envrionment variables within the portal (i use bulk edit)
3. restart the app <---- THIS IS IMPORTANT OTHERWISE THE INTERFACE DOESN'T RELOAD

**Reply by 0009fe3** [reply]
For me, the restart button in the Azure portal does not change the settings. But if I stop the web app and start it a few moments later again, changes to environment variables are honored.

**Comment by Darshan-Dholam** [comment]
how have yu deployed from locaally 
i am getting an issues with env variable
